src/react/AstarteClient.js

```diff
@@ -357,7 +379,7 @@ export default class AstarteClient {
     }
 
     const socketUrl = new URL(this.apiConfig.phoenixSocket(this.config));
-    socketUrl.protocol = (this.secureConnection ? "wss" : "ws")
+    socketUrl.protocol = (socketUrl.protocol === "https:" ? "wss" : "ws")
```

Comment:
Does the protocol include the `:`? It seems strange that `https:` does while `wss` and `ws` do not.

Reply:
Yes, the protocol getter returs the "protocol" part of the URL, including the colon.
The setter does the right thing, so there is no need to add it there.